### ISSUE TITLE
Train using custom dataset

### ISSUE THREAD

**Issue by z7r7y7** [aau-cns/poet#17]
Thank you for providing this excellent project! I would like to train on my custom dataset. In the _backbone.py_ file, I noticed that setting `self[0].train_backbone` to `True` results in a `NotImplementedError`. Does this mean that the backbone is currently not trainable? If I want to use Mask R-CNN as the backbone for training on my custom dataset, what steps should I follow?

**Comment by tgjantos** [comment]
Yes, currently PoET does not support the training of the backbone. We intended PoET to be an extension to any pre-trained backbone. 

If you want to use Mask R-CNN as the backbone for training on your custom dataset, you should first pre-train it separately on your dataset for object detection. Once you have the network trained you can include the pre-trained weights in the PoET training with the argument `--backbone_weights`.

Hope this helps you!

Best,
Thomas

**Comment by z7r7y7** [comment]
Thank you for your reply! I noticed that there are many versions of Mask R-CNN on GitHub. May I ask which version's weights can be directly loaded with the argument `--backbone_weights`?

**Comment by tgjantos** [comment]
You can check it out in the [backbone_maskrcnn.py](https://github.com/aau-cns/poet/blob/main/models/backbone_maskrcnn.py) file. As of now you can use the model how it is provided by [PyTorch](https://pytorch.org/vision/main/models/mask_rcnn.html) with a ResNet-50 backbone. 

However, you extend the code to use any object detector backbone you want as long as you return the necessary feature maps and detected object.

**Comment by z7r7y7** [comment]
You're right. Due to the presence of objects with similar shapes but varying sizes in my custom dataset, and the uncertainty of scale in monocular RGB images, relying solely on RGB images may yield suboptimal results. Therefore, I intend to fuse depth information with RGB information as input to the network.

**Comment by tgjantos** [comment]
I don't see any limitation regarding the transformer part to process a combination of RGB and depth feature maps. Therefore, if you have a backbone network that produces such feature maps, it should work out!

Let me know how it goes!

Best,
Thomas
